File: 20221012/모범답안/20221012_01/WordPatternGame.py
def solution(p, s):
    """
    :param p: str
    :param s: str
    :return: boolean
    """

    words = s.split(' ')
    map = {}

    if len(words) != len(p): return False
    if len(set(words)) != len(set(p)): return False

    for i in range(len(p)):
        if (p[i] in map) and map[p[i]] != words[i]: return False
        map[p[i]] = words[i]

    return True

File: 20221012/모범답안/20221012_01/WordPatternGame.js
/**
 * @param p {string}
 * @param s {string}
 * @returns {boolean}
 */
function solution(p, s) {
  const words = s.split(' ');
  const map = new Map();

  if (words.length !== p.length) return false;
  if (new Set(words).size !== new Set(p).size) return false;

  for (let i = 0; i < p.length; i++) {
    if (map.has(p[i]) && map.get(p[i]) !== words[i]) return false;
    map.set(p[i], words[i]);
  }
  return true;

}

solution